openmrs/src/main/java/org/motechproject/openmrs/tasks/OpenMRSTaskDataProvider.java

```diff
@@ -188,18 +189,22 @@ private Encounter getEncounter(String lookupName, Map<String, String> lookupFiel
     private Patient getPatient(String lookupName, Map<String, String> lookupFields, String configName) {
         Patient patient = null;
 
-        switch (lookupName) {
-            case BY_MOTECH_ID: patient = patientService.getPatientByMotechId(configName, lookupFields.get(MOTECH_ID));
-                break;
-            case BY_UUID: patient = patientService.getPatientByUuid(configName, lookupFields.get(UUID));
-                break;
-            default:
-                if (lookupFields != null) {
-                    patient = patientService.getPatientByIdentifier(configName, lookupFields.get(IDENTIFIER_ID), lookupName);
-                } else {
-                    LOGGER.error("Lookup with name {} doesn't exist for patient object", lookupName);
-                }
-                break;
+        try {
+            switch (lookupName) {
+                case BY_MOTECH_ID: patient = patientService.getPatientByMotechId(configName, lookupFields.get(MOTECH_ID));
+                    break;
+                case BY_UUID: patient = patientService.getPatientByUuid(configName, lookupFields.get(UUID));
+                    break;
+                default:
+                    if (lookupFields != null) {
+                        patient = patientService.getPatientByIdentifier(configName, lookupFields.get(IDENTIFIER_ID), lookupName);
+                    } else {
+                        LOGGER.error("Lookup with name {} doesn't exist for patient object", lookupName);
+                    }
+                    break;
+            }
+        } catch (OpenMRSException e) {
+            LOGGER.warn(e.getMessage());
```

Comment:
please use error level and log also whole stacktrace from exception (use exception object as a second argument)